controllers/applicationvolumereplication_controller.go

```diff
@@ -125,6 +126,42 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
 	return ctrl.Result{Requeue: requeue}, nil
 }
 
+func IsManifestInAppliedState(mw *ocmworkv1.ManifestWork) bool {
+	isApplied := false
+	conditions := mw.Status.Conditions
+
+	if len(conditions) > 0 {
+		// sort conditions by timestamp. Index 0 = most recent
+		sort.Slice(conditions, func(a, b int) bool {
+			return conditions[b].LastTransitionTime.Before(&conditions[a].LastTransitionTime)
+		})
+
+		mostRecentTimestamp := conditions[0].LastTransitionTime
+
+		// loop through conditions until not in the most recent one anymore
+		for index := range conditions {
+			// only index 1+ needs exit check
+			if index > 0 {
+				timestamp := conditions[index].LastTransitionTime
+
+				// only evaluate conditions on most recent timestamp
+				if timestamp != mostRecentTimestamp {
+					break
+				}
+			}
+
+			// check if state is applied
+			if conditions[index].Type == "Applied" {
```

Comment:
I think we should be looking at the condition.Type and also the condition.Status [true|false|unknown] here to determine if work is applied or not.

Also, if conditions have "Applied" and "Degraded" with the same time stamp and status as `true`, we should call it not applied?

Going by [generateUpdateStatusFunc](https://github.com/open-cluster-management/work/blob/11f530f467683e5b8d6daaff80d32fad97cf54b8/pkg/spoke/controllers/manifestcontroller/manifestwork_controller.go#L268-L296)

Reply:
I updated `IsManifestInAppliedState` to check for both the Condition.Type and the Condition.Status = ConditionTrue on `d41ab23fee18f7e5e248be22830fc12385e9075e`. I hadn't considered "Degraded", but rolled that into the logic . Now, if the most recent timestamps contain an "Applied" and a "Degraded" Type, the function will return false.